### PR TITLE
Nullable changes for boolean type

### PR DESCRIPTION
**Pull request by buyaa-n**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Boolean.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@
 using System.Runtime.CompilerServices;
 using System.Runtime.Versioning;
 
+#nullable enable
```
Comment (safern) [inline src/System.Private.CoreLib/shared/System/Boolean.cs]: Could you move this to before the `using` statements, to be consistent to what the tool now produces?

File: src/System.Private.CoreLib/shared/System/Boolean.cs [hunk 2]
```diff
@@ -306,79 +307,80 @@ public TypeCode GetTypeCode()
...
         }
 
-        object IConvertible.ToType(Type type, IFormatProvider provider)
+        object IConvertible.ToType(Type type, IFormatProvider? provider)
         {
             return Convert.DefaultToType((IConvertible)this, type, provider);
         }
     }
 }
+#nullable restore
```
Comment (safern) [inline src/System.Private.CoreLib/shared/System/Boolean.cs]: This is not needed, my bad for the wrong guidance, could you please remove it?

Comment (buyaa-n) [reply src/System.Private.CoreLib/shared/System/Boolean.cs]: sure

File: src/System.Private.CoreLib/shared/System/Boolean.cs (hunk 1 above)
Comment (buyaa-n) [reply src/System.Private.CoreLib/shared/System/Boolean.cs]: ok

**Pull request merged**